### PR TITLE
fix: Html encode query string

### PR DESCRIPTION
**Pull request by pdavies88**

### CODE REVIEW

File: src/assets/javascripts/components/search.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ var searchOnResultsPage = function() {
   if (typeof searchTerm != "undefined"){
     desktop_search_input.value = searchTerm;
     mobile_search_input.value = searchTerm;
-    document.getElementById('query-term').innerHTML = searchTerm;
+    document.getElementById('query-term').innerText = htmlEncode(searchTerm);
```
Comment (k80bowman) [inline src/assets/javascripts/components/search.js:69]: should we be encoding it before it gets put into the search input as well? (lines 67 & 68)

Comment (andresriveratoro) [reply src/assets/javascripts/components/search.js:69]: It looks like we may want to do that.
https://cheatsheetseries.owasp.org/cheatsheets/DOM_based_XSS_Prevention_Cheat_Sheet.html#rule-2-javascript-escape-before-inserting-untrusted-data-into-html-attribute-subcontext-within-the-execution-context

**Comment by pdavies88** [comment]
Following guidance from: https://portswigger.net/web-security/cross-site-scripting/preventing

File: src/assets/javascripts/components/search.js (hunk 1 above)
Comment (k80bowman) [reply src/assets/javascripts/components/search.js:69]: Could you save `htmlEncode(searchTerm)` to a variable instead of calling it three times?

Comment (pdavies88) [reply src/assets/javascripts/components/search.js]: Good call just updated with that

**Review by k80bowman — APPROVED** [review]
Looks great, thank you!

**Pull request merged**